ksp_plugin/plugin.cpp

```diff
@@ -694,5 +758,75 @@ void Plugin::EvolveProlongationsAndBubble(Instant const& t) {
   }
 }
 
+void Plugin::UpdatePredictions() {
+  clear_predictions();
+  if (has_predicted_vessel()) {
+    NBodySystem<Barycentric>::Trajectories predictions;
+    predictions.reserve(celestials_.size() + 1);
+    for (auto const& index_celestial : celestials_) {
+      auto const& celestial = index_celestial.second;
+      auto const inserted =
+          system_predictions_.emplace(
+              celestial.get(),
+              celestial->mutable_prolongation()->NewFork(
+                  celestial->prolongation().last().time()));
+      CHECK(inserted.second);
+      not_null<Trajectory<Barycentric>*> const trajectory =
+          inserted.first->second;
+      predictions.emplace_back(trajectory);
+    }
+    prediction_ = predicted_vessel_->mutable_prolongation()->NewFork(
+                      predicted_vessel_->prolongation().last().time());
+    predictions.emplace_back(prediction_);
+    n_body_system_->Integrate(
+        prolongation_integrator_,
+        current_time_ + prediction_length_,
+        prediction_step_,
+        1,  // sampling_period
+        false,  // tmax_is_exact
+        predictions);
+  }
+}
+
+RenderedTrajectory<World> Plugin::RenderTrajectory(
+    Trajectory<Barycentric> const& actual_trajectory,
+    Trajectory<Barycentric>::TransformingIterator<Rendering> const& actual_it,
+    not_null<Transforms<Barycentric, Rendering, Barycentric>*> const transforms,
+    Position<World> const& sun_world_position) const {
+  RenderedTrajectory<World> result;
+  auto const to_world =
```

Comment:
You lost a comment that started with "First..." around here.  Please restore, the comments read weird now.